### DISCUSSION TITLE
Just a thought

### DISCUSSION THREAD

**Discussion by NightXof** [HarryDulaney/intro-to-java-programming#59] [Ideas]
Hey Harry, It's a wonderful thing that you have wrote all code and documented it so well that i have bookmarked your repository!. Now that I've sugarcoated my only gripe is the exercises in the each chapter are given with such concepts in mind where we are only limited to the areas they are covered. For example the deck of cards exercise in the chapter 3 - Selections, the exercise is supposed to be done only with selections statements, logical and conditional expressions, but you have used arrays which is not gonna come until chapter 6. It was kind of a bummer but yeah i wrote my own code for it before looking at yours and I found this mismatch. Thanks if you have read so far.

**Comment by HarryDulaney** [comment]
Hey, appreciate your feedback @NightXof that's a good point for sure. I would like to restrict solutions to only concepts that have been introduced up until they point. Although in the case of Arrays they are so core to programming I don't see a benefit to avoiding them until they are specifically covered. I'd say too that the first 8 chapters are a learning block on their own so if we restrict concepts it should be by a group of chapters. That being said, I'm open to making updates to existing solutions for sure. I see your point on deck of cards, that's probably supposed to prompt the solver to practice creating complex if else logic so the array solution kind of misses the point. Please feel free to open a PR with your solution!

**Reply by HarryDulaney** [reply]
Hey @NightXof, I updated that deck of cards exercise, let me know if you see any more improvements.

**Reply by NightXof** [reply]
Thanks a lot man, sorry life got busy again..

**Comment by HarryDulaney** [comment]
> Hey Harry, It's a wonderful thing that you have wrote all code and documented it so well that i have bookmarked your repository!. Now that I've sugarcoated my only gripe is the exercises in the each chapter are given with such concepts in mind where we are only limited to the areas they are covered. For example the deck of cards exercise in the chapter 3 - Selections, the exercise is supposed to be done only with selections statements, logical and conditional expressions, but you have used arrays which is not gonna come until chapter 6. It was kind of a bummer but yeah i wrote my own code for it before looking at yours and I found this mismatch. Thanks if you have read so far.

Thanks again for the suggestions, I updated the exercises in Chapter 3 to use selection statements instead of arrays per chapter curriculum.
